xivo/asterisk/protocol_interface.py

```diff
@@ -22,46 +22,33 @@ def __init__(self, invalid_channel: str | None = None) -> None:
 
 
 def protocol_interface_from_channel(channel: str) -> ProtocolInterface:
-    matches = channel_regexp.search(channel)
-    if matches is None:
+    if (match := channel_regexp.search(channel)) is None:
         raise InvalidChannelError(channel)
 
-    protocol = matches.group(1)
-    interface = matches.group(2)
-
-    if protocol == 'pjsip':
-        protocol = 'sip'
-    elif protocol == 'PJSIP':
-        protocol = 'SIP'
-
+    protocol, interface = match.groups()
+    if protocol.lower() == 'pjsip':
+        protocol = protocol[-3:]
```

Comment:
> > Does the case matter?
> 
> I don't know smile (in theory I don't think, but it's used so much everywhere that I cannot guarantee)

Haha, that was my worry too. That's why I thought it was safest to use the original string then the case is preserved. I don't see why it would though.

Reply:
What do you think of:
```suggestion
        protocol = 'SIP' if protocol.isupper() else 'sip'
```
It made the code more explicit and we don't want to handle 'SiP' anyways

Reply:
Sure, I like that thanks. Explicit, but not redundant.